### PR TITLE
Increase coverage of the example program

### PR DESCRIPTION
**Pull request by johnsonjh**
* More coverage for the example program
* Minimal changes and all linters are happy
* Overall a +0.2% gain to 90.195%

### CODE REVIEW

File: example/example.c [hunk 1]
```diff
@@ -44,7 +45,13 @@ int report_error(void);
  * @returns EXIT_SUCCESS if execution completes successfully, or EXIT_FAILURE
  * if an error occurs.
  */
-int main(void) {
+int main(int argc, char** argv) {
+#if defined(__HAIKU__) && defined(NDEBUG)
+    disable_debugger(1);
```
Comment (aremmell) [inline example/example.c:50]: what does disaable_debugger do, besides disable the debugger?

File: example/example.c [hunk 2]
```diff
@@ -88,7 +95,7 @@ int main(void) {
      */
     sirfileid fileid = sir_addfile("libsir-example.log", SIRL_ALL, SIRO_NONAME | SIRO_NOHOST);
     if (0 == fileid)
-        report_error();
+        (void)report_error();
```
Comment (aremmell) [inline example/example.c:98]: At this point with this many (void) in front of it, might as well nuke the return type and just not use it as the return value.

File: example/example.c [hunk 3]
```diff
@@ -139,13 +146,13 @@ int main(void) {
 
 #if !defined(SIR_NO_SYSTEM_LOGGERS)
```
Comment (aremmell) [inline example/example.c:147]: Wouldn't this decrease coverage? We want to hit those exports even with NO_SYSTEM_LOGGERS, yeah?

**Review by aremmell — DISMISSED** [review]
tbh I would rather have a lower coverage score than to muddy the example with preprocs and stuff... I dunno, I'll have to see the merged file. Maybe we can put this in another branch for now called say gcovr, and work on coverage there and cherry-pick the stuff we really like into master?

File: example/example.c (hunk 1 above)
Comment (johnsonjh) [reply example/example.c:50]: That's it.  If you don't call that, then by default a graphical debugger interface is attached to any crashing program.  It's nice if you are using the system, but if you connect with SSH or running via a script, the process will just appear to hang forever, no visible crash or core file or anything.

File: example/example.c (hunk 3 above)
Comment (johnsonjh) [reply example/example.c:147]: That's unchanged from the original.  I didn't try to change that (yet).

Comment (aremmell) [reply example/example.c:147]: Yeah, we should remove that. Even though we hit those paths during testing, can't hurt and makes the .c look less terrible.

**Comment by johnsonjh** [comment]
> tbh I would rather have a lower coverage score than to muddy the example with preprocs and stuff... I dunno, I'll have to see the merged file. Maybe we can put this in another branch for now called say gcovr, and work on coverage there and cherry-pick the stuff we really like into master?

I'll do something more minimal, tell me what you think.

**Comment by johnsonjh** [comment]
Clearer and doesn't upset the other linters...

**Comment by johnsonjh** [comment]
@aremmell ^

**Review by johnsonjh — COMMENTED** [review]
Moved other changes to new PR

**Review by aremmell — COMMENTED** [review]
tbh I'd rather leave the example alone for 0.2%–all I care about is the green :)

**Comment by johnsonjh** [comment]
Well we're only 0.6 or something into the green! :)

**Comment by johnsonjh** [comment]
@aremmell I'll close this out - I have a better idea

**Pull request closed without merging**